packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -1136,6 +1149,11 @@ export class FluidDataStoreRuntime
 	 * This typically happens when we reconnect and there are unacked messages.
 	 * @param content - The content of the original message.
 	 * @param localOpMetadata - The local metadata associated with the original message.
+	 *
+	 * @privateRemarks
+	 * `type` parameter's type of `DataStoreMessageType` is a covariance exception
+	 * over `string` that `IFluidDataStoreChannel` specifies. (`unreachableCase`
```

Comment:
Some thoughts on this - It's strictly true, but only messages that were submitted from here will come back for resubmit. So IMO it is a legitimate, stable and robust way to type this. I don't think the warning about unreachableCase being hit over time is warranted.

I believe it's correct that `IFluidDataStoreChannel` should specify `string` here, since another implementation could have its own schema. And we haven't wanted to parametrize everything, it leads to quite a soup of generics and doesn't provide much value in practice IMO.